.storybook/helpers.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react'; // eslint-disable-line no-unused-vars
+import { withKnobs, boolean } from '@storybook/addon-knobs';
+import wrapperManifest from './../blocks/init/wrapper/manifest.json';
+import {
+  BlockEditorKeyboardShortcuts,
+  BlockEditorProvider,
+  BlockList,
+  BlockInspector,
+  WritingFlow,
+  ObserveTyping,
+} from '@wordpress/block-editor';
+import {
+  Popover,
+  SlotFillProvider,
+  DropZoneProvider,
+} from '@wordpress/components';
+
+/**
+ * Generate Blocks Wrapper.
+ *
+ * @param {bool} isActive Set if Wrapper is active or not.
+ */
+export const hasWrapper = (isActive) => {
+  const wrapper = {
+    hasWrapper: true,
+    styleContentWidthLarge: wrapperManifest.attributes.styleContentWidthLarge.default,
+    styleContentOffsetLarge: wrapperManifest.attributes.styleContentOffsetLarge.default,
+    styleContainerWidthLarge: wrapperManifest.attributes.styleContainerWidthLarge.default,
+    styleContainerSpacingLarge: wrapperManifest.attributes.styleContainerSpacingLarge.default,
+    styleSpacingBottomLarge: wrapperManifest.attributes.styleSpacingBottomLarge.default,
+    styleHideBlockLarge: wrapperManifest.attributes.styleHideBlockLarge.default,
+    styleHideBlockDesktop: wrapperManifest.attributes.styleHideBlockDesktop.default,
+  };
+
+  return isActive ? wrapper : {};
+}
+
+/**
+ * Define if block has wrapper decorator. Used in storybook knobs.
+ *
+ * @param {json} manifest Block Manifest data.
+ */
+export const hasWrapperDecorator = (manifest) => {
+  if ( typeof manifest.hasWrapper === 'undefined' ) {
+    manifest.hasWrapper = true;
+  }
+
+  return (manifest.hasWrapper ? { decorators: [withKnobs] } : '')
+};
+
+/**
+ * Load actial Gutenberg and all the magic.
+ *
+ * @param {object} props All Props for blocks.
+ */
+export const Gutenberg = (props) => {
+  const {
+    props: {
+      blocks,
+      useWrapper = false,
+    },
+  } = props;
+
+  const blocksProps = blocks.map((block) => {
+    return {
+      ...block,
+      attributes: {...block.attributes, ...hasWrapper(boolean('Use Wrapper', useWrapper))},
+      clientId: id(),
+      isValid: true,
+    }
+  });
+
+  return (
+    <div className="playground">
+      <SlotFillProvider>
+        <DropZoneProvider>
+          <BlockEditorProvider
+            value={ blocksProps }
+          >
+            <div className="playground__sidebar">
+              <BlockInspector />
+            </div>
+            <div className="editor-styles-wrapper">
+              <BlockEditorKeyboardShortcuts />
+              <WritingFlow>
+                <ObserveTyping>
+                  <BlockList />
+                </ObserveTyping>
+              </WritingFlow>
+            </div>
+            <Popover.Slot />
+          </BlockEditorProvider>
+        </DropZoneProvider>
+      </SlotFillProvider>
+    </div>
+  );
+};
+
+/**
+ * Define generic block ID.
+ */
+export const id = () => (Math.random() * Math.floor(10)).toString(36).substring(2);
```

Comment:
Maybe to avoid hoisting, this should be declared before Gutenberg as block props returns id()

Reply:
👍